.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs

```diff
@@ -368,15 +368,27 @@ private List<Token> MatchYearPeriod(string text, DateObject referenceDate)
                 {
                     var yearMatches = this.config.YearRegex.Matches(match.Value);
                     var allDigitYear = true;
+                    var isValidYear = true;
 
                     foreach (Match yearMatch in yearMatches)
                     {
-                        if (yearMatch.Length != Constants.FourDigitsYearLength)
+                        var year = config.DatePointExtractor.GetYearFromText(yearMatch);
+                        if (!(year >= Constants.MinYearNum && year <= Constants.MaxYearNum))
+                        {
+                            isValidYear = false;
+                            break;
```

Comment:
Why this break? Then you are skipping all other matches in yearMatches.

Reply:
Does this call to YearRegex happen for every period with a year? Or only for "year-only" cases?
If the first, then it may also introduce other issues.

Reply:
If one year is invalid in a year period, i think we needn't to check another one

Reply:
There may be more than one year involved in a sentence, no?

Reply:
And if you do this ```break```, then the ```if/continue``` never happen.

Reply:
I just break the loop of "yearMatches", don't check if another year is valid. why if/continue never happen..If/continue is out of the loop.

Reply:
That's what happens when people review on a tablet :)